flake.nix

```diff
@@ -6,101 +6,28 @@
-          goPkgs = with pkgs; [
-            go
-            gotools
-            go-migrate
-          ];
-        in
-        {
-          pre-commit = {
-            check.enable = true;
-            pkgs = pkgs;
-            settings = {
-              hooks = {
-                gofmt.enable = true;
-                govet.enable = true;
-                golangci-lint = {
-                  enable = true;
-                  name = "golangci-lint";
-                  description = "Lint my golang code";
-                  files = "\.go$";
-                  entry =
-                    let
-                      script = pkgs.writeShellScript "precommit-golangci" ''
-                        set -e
-                        export PATH=$PATH:${pkgs.go}/bin
-                        ${pkgs.golangci-lint}/bin/golangci-lint run --new-from-rev HEAD --fix
-                      '';
-                    in
-                    builtins.toString script;
-                  require_serial = true;
-                  pass_filenames = false;
-                };
-                goimports = {
-                  enable = true;
-                  name = "goimports";
-                  description = "Format my golang code";
-                  files = "\.go$";
-                  entry =
-                    let
-                      script = pkgs.writeShellScript "precommit-goimports" ''
-                        set -e
-                        failed=false
-                        for file in "$@"; do
-                            # redirect stderr so that violations and summaries are properly interleaved.
-                            if ! ${pkgs.gotools}/bin/goimports -l -d "$file" 2>&1
-                            then
-                                failed=true
-                            fi
-                        done
-                        if [[ $failed == "true" ]]; then
-                            exit 1
-                        fi
-                      '';
-                    in
-                    builtins.toString script;
-                };
-              };
-            };
-          };
-
-          formatter = pkgs.nixpkgs-fmt;
-
-          devShells.default = pkgs.mkShell {
-            PLAYWRIGHT_BROWSERS_PATH = pkgs.playwright-driver.browsers;
-            shellHook = ''
-              ${config.pre-commit.installationScript}
-              echo 1>&2 "Oxidrive development shell, pre-commits are enabled by default and can be run using nix flake check!"
-            '';
-
-            packages = with pkgs; [
-              act
-              air
-              dioxus-cli
-              just
-              nodejs_20
-              rust
-              gh
-
-              # Server
-              goPkgs
-            ] ++ config.pre-commit.settings.enabledPackages;
-          };
+      perSystem = { system, ... }: {
+        rust = {
+          toolchain.file = ./web/rust-toolchain.toml;
+          overlay = import rust-overlay;
```

Comment:
Why don't we move this in the dedicated rust file?

Reply:
We still need to inject the rust-overlay in some ways (e.g. by making `nix/rust.nix` a function that returns the flake module). The toolchain file can be hardcoded as `../web/rust-toolchain.toml` if we want, but i felt it was really ugly :)